crates/sui-core/src/authority/authority_store.rs

```diff
@@ -1030,11 +1146,54 @@ impl AuthorityStore {
 
     /// Gets ObjectLockInfo that represents state of lock on an object.
     /// Returns UserInputError::ObjectNotFound if cannot find lock record for this object
-    pub(crate) fn get_lock(&self, obj_ref: ObjectRef, epoch_id: EpochId) -> SuiLockResult {
+    pub(crate) fn get_lock(
+        &self,
+        obj_ref: ObjectRef,
+        epoch_store: &AuthorityPerEpochStore,
+    ) -> SuiLockResult {
+        if epoch_store.object_lock_split_tables_enabled() {
+            self.get_lock_v2(obj_ref, epoch_store)
+        } else {
+            self.get_lock_v1(obj_ref, epoch_store.epoch())
+        }
+    }
+
+    fn get_lock_v2(
+        &self,
+        obj_ref: ObjectRef,
+        epoch_store: &AuthorityPerEpochStore,
+    ) -> SuiLockResult {
+        if self
```

Comment:
Not necessary - the mutexes are only needed to prevent a race when we are doing a test-and-set operation (as in acquire_locks)

Reply:
Yeah, you are right. Probably not necessary in this case. I was thinking that it was reading at a consistent snapshot before, but now it may not (we read live_owned_object_markers first, then a lot of things happen in between that could update live_owned_object_markers/owned_object_locked_transactions, and then we read owned_object_locked_transactions). But given that now we don't delete owned_object_locked_transactions anymore, I think it's ok.